packages/api/app/index.js

```diff
@@ -13,10 +13,38 @@ const awsServerlessExpressMiddleware = require('aws-serverless-express/middlewar
 
 const { services } = require('@cumulus/aws-client');
 const { MissingRequiredEnvVar } = require('@cumulus/errors');
+const { secretsManager } = require('@cumulus/aws-client/services');
+const Logger = require('@cumulus/logger');
 
 const router = require('./routes');
 const { jsonBodyParser } = require('./middleware');
 
+const log = new Logger({ sender: '@api/index' });
+
+// Load Environment Variables
+// This should be done outside of the handler to minimize Secrets Manager calls.
+const initEnvVarsFunction = async () => {
+  if (!process.env.api_config_secret_id) {
+    throw new Error('A configuration secret must be defined in process.env.api_config_secret_id for API to function');
+  }
+  try {
+    const response = await secretsManager().getSecretValue(
+      { SecretId: process.env.api_config_secret_id }
+    ).promise();
+    if (response.SecretString === undefined) {
+      throw new Error(`AWS Secret did not contain a stored value: ${process.env.api_config_secret_id}`);
+    }
+    const envSecret = JSON.parse(response.SecretString);
+    log.warn('Setting env vars');
+    log.warn(envSecret);
+    process.env = { ...envSecret, ...process.env };
+  } catch (error) {
+    log.error(`Encountered error trying to set environment variables from secret ${process.env.api_config_secret_id}`, error);
```

Comment:
```suggestion
    log.error(`Encountered error trying to set environment variables from secret ${process.env.api_config_secret_id}`, error);
    throw error;
```

We still want to re-throw the error here I believe. I'd say we need a unit test to catch this and ensure that the API/handler throws an error or returns an error response when this is reached

Reply:
Really I think we should just write some unit tests of `initEnvVarsFunction` itself to prove that it throws / succeeds depending on the test setup. We already have the handler unit test

Reply:
Agreed.   They're just a bit annoying in organization without manipulating the package cache as separate files are required. 

0fa373320